### PR TITLE
Tests: Fix nil pointer deref in expansion test

### PR DESCRIPTION
**Pull request by alromeros**
### What this PR does

This PR aims to fix a nil pointer dereference in a volume expansion test.

### Release note

```release-note
NONE
```

### CODE REVIEW

**Comment by alromeros** [comment]
/cc @akalenyu 
Decided not to include https://github.com/kubevirt/kubevirt/pull/12225 since we still need to access the volume status and as you pointed out, it would be clumsy to backport.

File: tests/storage/datavolume.go [hunk 1]
```diff
@@ -281,7 +281,21 @@ var _ = SIGDescribe("DataVolume Integration", func() {
...
+				Eventually(func() bool {
+					vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(context.Background(), vmi.Name, metav1.GetOptions{})
+					Expect(err).ToNot(HaveOccurred())
+					for _, volStatus := range vmi.Status.VolumeStatus {
+						if volStatus.Name == "disk0" {
+							volumeStatus = volStatus
+							return true
+						}
+					}
+					return false
```
Comment (akalenyu) [inline tests/storage/datavolume.go:294]: maybe instead of the true/false dance just return the field you're interested in? or the entire volumestatus?

Comment (alromeros) [reply tests/storage/datavolume.go:294]: I think the code would remain almost the same in terms of length/readability, as we would need to assign it elsewhere even if we return the volume status.

Comment (akalenyu) [reply tests/storage/datavolume.go:294]: Yeah but you're going to have to deal with an obfuscated error if this ever fails ("expected false to be true")

Comment (alromeros) [reply tests/storage/datavolume.go:294]: Added a message in the `eventually` so we get a more verbose error if it fails.

**Review by akalenyu — APPROVED** [review]

File: tests/storage/datavolume.go [hunk 2]
```diff
@@ -281,7 +281,21 @@ var _ = SIGDescribe("DataVolume Integration", func() {
...
+				}, 30*time.Second, time.Second).Should(BeTrue(), "Expected VolumeStatus for 'disk0' to be available")
+
+				Expect(volumeStatus.PersistentVolumeClaimInfo.FilesystemOverhead).ToNot(BeNil())
```
Comment (iholder101) [inline tests/storage/datavolume.go:297]: `volumeStatus.PersistentVolumeClaimInfo` might be nil as well

```suggestion
				Expect(volumeStatus.PersistentVolumeClaimInfo).ToNot(BeNil())
				Expect(volumeStatus.PersistentVolumeClaimInfo.FilesystemOverhead).ToNot(BeNil())
```

Comment (alromeros) [reply tests/storage/datavolume.go:298]: ack

**Pull request merged**